runtime/zeitgeist/src/integration_tests/xcm/tests/transfers.rs

```diff
@@ -0,0 +1,336 @@
+
+        // Verify that fees (of foreign currency) have been put into treasury
+        assert_eq!(
+            Tokens::free_balance(FOREIGN_ZTG_ID, &ZeitgeistTreasuryAccount::get()),
+            treasury_initial_balance + ztg_fee()
+        )
+    });
+}
+
+#[test]
+fn transfer_ztg_sibling_to_zeitgeist() {
+    TestNet::reset();
+
+    // In order to be able to transfer ZTG from Sibling to Zeitgeist, we need to first send
+    // ZTG from Zeitgeist to Sibling, or else it fails since it'd be like Sibling had minted
+    // ZTG on their side.
+    transfer_ztg_to_sibling();
+
+    let alice_initial_balance = ztg(5);
+    let bob_initial_balance = ztg(5) - ztg_fee();
+    let mut treasury_initial_balance = 0;
+    let sibling_sovereign_initial_balance = ztg(5);
+    let transfer_amount = ztg(1);
+    // Note: This asset was registered in `transfer_ztg_to_sibling`
+
+    Zeitgeist::execute_with(|| {
+        treasury_initial_balance = Balances::free_balance(ZeitgeistTreasuryAccount::get());
+
+        assert_eq!(Balances::free_balance(&ALICE.into()), alice_initial_balance);
+        assert_eq!(Balances::free_balance(&sibling_account()), sibling_sovereign_initial_balance);
+    });
+
+    Sibling::execute_with(|| {
+        assert_eq!(Balances::free_balance(&zeitgeist_account()), 0);
+        assert_eq!(Tokens::free_balance(FOREIGN_ZTG_ID, &BOB.into()), bob_initial_balance);
+        assert_ok!(XTokens::transfer(
+            Origin::signed(BOB.into()),
+            FOREIGN_ZTG_ID,
+            transfer_amount,
+            Box::new(
+                MultiLocation::new(
+                    1,
+                    X2(
+                        Parachain(zeitgeist::ID),
+                        Junction::AccountId32 { network: NetworkId::Any, id: ALICE.into() }
+                    )
+                )
+                .into()
+            ),
+            4_000_000_000,
+        ));
+
+        // Confirm that Bobs's balance is initial balance - amount transferred
+        assert_eq!(
+            Tokens::free_balance(FOREIGN_ZTG_ID, &BOB.into()),
+            bob_initial_balance - transfer_amount
+        );
+    });
+
+    Zeitgeist::execute_with(|| {
+        // Verify that ALICE now has initial balance + amount transferred - fee
+        assert_eq!(
+            Balances::free_balance(&ALICE.into()),
+            alice_initial_balance + transfer_amount - ztg_fee(),
+        );
+
+        // Verify that the reserve has been adjusted properly
+        assert_eq!(
+            Balances::free_balance(&sibling_account()),
+            sibling_sovereign_initial_balance - transfer_amount
+        );
+
+        // Verify that fees (of native currency) have been put into treasury
+        assert_eq!(
+            Balances::free_balance(ZeitgeistTreasuryAccount::get()),
+            treasury_initial_balance + ztg_fee()
+        )
+    });
+}
+
+#[test]
+fn transfer_ksm_from_relay_chain() {
+    let transfer_amount: Balance = ksm(1);
```

Comment:
We did not touch any of the initial ksm balances and the sovereign account of the relay chain is also clean. There is no need to reset the testnet. However, this might become necessary if additional tests in regards to the interaction between the native and the relay chains are added.

Reply:
It is not necessary because that intermediate state changes that can happen in other tests don't influence the correctness of this test. However, I am in favor of using a fail-safe approach that does not require so much thinking :smile:
Added in https://github.com/zeitgeistpm/zeitgeist/pull/661/commits/b2c1e1f3f41d3ada12db337ff502b9181896f44f

Reply:
https://github.com/zeitgeistpm/zeitgeist/pull/661/commits/fdfa04bbe948929dcbe30a2f349641c3dc182899